test/encryption/test_encryption.py

```diff
@@ -0,0 +1,160 @@
+    pdf.set_font("helvetica", size=12)
+    pdf.cell(txt="hello world")
+    pdf.set_encryption(owner_password="fpdf2", permissions=AccessPermission.all())
+    assert_pdf_equal(pdf, HERE / "encryption_rc4.pdf", tmp_path)
+
+
+def test_encryption_rc4_permissions(tmp_path):
+    pdf = FPDF()
+    pdf.set_creation_date(EPOCH)
+
+    def custom_file_id():
+        return pdf._default_file_id(bytearray([0xFF]))
+
+    pdf.file_id = custom_file_id
+    pdf.set_author("author")
+    pdf.set_subject("string to be encrypted")
+    pdf.add_page()
+    pdf.set_font("helvetica", size=12)
+    pdf.cell(txt="hello world")
+    pdf.set_encryption(
+        owner_password="fpdf2",
+        permissions=AccessPermission.PRINT_LOW_RES | AccessPermission.PRINT_HIGH_RES,
+    )
+    assert_pdf_equal(pdf, HERE / "encryption_rc4_permissions.pdf", tmp_path)
+
+
+def test_no_encryption(tmp_path):
+    pdf = FPDF()
+    pdf.set_creation_date(EPOCH)
+
+    def custom_file_id():
+        return pdf._default_file_id(bytearray([0xFF]))
+
+    pdf.file_id = custom_file_id
+    pdf.set_author("author")
+    pdf.set_subject("string to be encrypted")
+    pdf.add_page()
+    pdf.set_font("helvetica", size=12)
+    pdf.cell(txt="hello world")
+    pdf.set_encryption(
+        owner_password="fpdf2",
+        encryption_method=EncryptionMethod.NO_ENCRYPTION,
+        permissions=AccessPermission.none(),
+    )
+    assert_pdf_equal(pdf, HERE / "no_encryption.pdf", tmp_path)
+
+
+def test_encryption_rc4_user_password(tmp_path):
+    pdf = FPDF()
+    pdf.set_creation_date(EPOCH)
+
+    def custom_file_id():
+        return pdf._default_file_id(bytearray([0xFF]))
+
+    pdf.file_id = custom_file_id
+    pdf.set_author("author")
+    pdf.set_subject("string to be encrypted")
+    pdf.add_page()
+    pdf.set_font("helvetica", size=12)
+    pdf.cell(txt="hello world")
+    pdf.set_encryption(
+        owner_password="fpdf2",
+        user_password="654321",
+        permissions=AccessPermission.PRINT_LOW_RES | AccessPermission.PRINT_HIGH_RES,
+    )
+    assert_pdf_equal(pdf, HERE / "encryption_rc4_user_password.pdf", tmp_path)
+
+
+def test_encryption_aes128(tmp_path):
+    pdf = FPDF()
+    pdf.set_creation_date(EPOCH)
+
+    def custom_file_id():
+        return pdf._default_file_id(bytearray([0xFF]))
+
+    pdf.file_id = custom_file_id
+
+    def fixed_iv(size):
+        return bytearray(size)
+
+    pdf.set_author("author")
+    pdf.set_subject("string to be encrypted")
+    pdf.add_page()
+    pdf.set_font("helvetica", size=12)
+    pdf.cell(txt="hello world")
+    pdf.set_encryption(
+        owner_password="fpdf2",
+        encryption_method=EncryptionMethod.AES_128,
+        permissions=AccessPermission.none(),
+    )
+    pdf._security_handler.get_initialization_vector = fixed_iv
+    assert_pdf_equal(pdf, HERE / "encryption_aes128.pdf", tmp_path)
+
+
+def test_encrypt_metadata(tmp_path):
+    pdf = FPDF()
+    pdf.set_creation_date(EPOCH)
+
+    def custom_file_id():
```

Comment:
Could some unit tests **not** define a `custom_file_id`?
If it is mandatory to define one, it should be mentioned in the method docstring & in `docs/encryption.md`

Reply:
It is not mandatory, however a different file_id will change the contents of the file as that's part of the encryption key.
I need custom file id's to keep the encryption key always the same so we can compare test results.

Reply:
nevermind... file id should be stable enough if the creation date is the same. I removed custom id from 2 tests.

Reply:
Perfect, thank you!